core_v2/sources/price_model.move

```diff
@@ -0,0 +1,121 @@
+module aptos_names_v2::price_model {
+    use aptos_names_v2::config;
+    use aptos_std::math64;
+    use std::error;
+
+
+    /// The domain length is too short- currently the minimum is 2 characters
+    const EDOMAIN_TOO_SHORT: u64 = 1;
+    const SECONDS_PER_YEAR: u64 = 60 * 60 * 24 * 365;
+
+    /// The longer the name is registered for, the more expensive it is per year.
+    /// The curve is exponential, with every year costing more than the previous
+    fun scale_price_for_years(price: u64, years: u8): u64 {
+        // TODO: THIS WHOLE FUNCTION IS A PLACEHOLDER
+        let final_price = 0;
+        let multiplier = 100;
+        let years = (years as u64);
+        let i = 1;
+        while (i <= years) {
+            final_price = final_price + (price * multiplier) / 100;
+            multiplier = multiplier + 5 + i * 5;
+            i = i + 1;
+        };
+        final_price
+    }
+
+    /// There is a fixed cost per each tier of domain names, from 2 to >=6, and it also scales exponentially with number of years to register
+    public fun price_for_domain_v1(domain_length: u64, registration_secs: u64): u64 {
```

Comment:
these `v1` are distracting lol

I would just make it a fixed price and kill all this code about dynamic prices....